R/with.R

```diff
@@ -24,9 +24,34 @@ modify_defaults <- function(defaults, ...) {
   }
   vals <- list(...)
   nms <- names2(vals)
+  if ("default" %in% nms) {
+    sys_calls <- sys.calls()
+    prev_call_id <- length(sys_calls) - 1L
+    # getting default= is a lot harder for modify_defaults() because defaults= is a required
+    #   argument. so assume that this was intentional and restrict focus to linters_with_defaults calls
+    if (
+      prev_call_id > 0L && (
+        (is.name(prev_call <- sys_calls[[prev_call_id]][[1L]]) && identical(prev_call, quote(linters_with_defaults))) ||
+        (is.call(prev_call) && identical(prev_call, quote(lintr::linters_with_defaults)))
+      )
+    ) {
+      warning(
+        "'default' is not an argument to linters_with_defaults(). Did you mean 'defaults'? ",
+        "This warning will be removed when with_defaults() is fully deprecated."
+      )
+      defaults <- vals$default
+      vals$default <- NULL
+      valid_idx <- nms != "default"
+      nms <- nms[valid_idx]
```

Comment:
Is there a way to use `Recall()` after the repair to keep the change localized?

Also shouldn't do this if there is also a `defaults` argument.

Reply:
Didn't we eliminate `Recall()` as viable for the other similar checks in `lint()`/`lint_package()` because of headaches with `...`?

> Also shouldn't do this if there is also a defaults argument.

OK, actually that will make things a lot easier

Reply:
It would definitely be cleaner to put the warning in `linters_with_defaults()` though...
Shouldn't something like this work?

```r
dots <- list(...)
if (missing(defaults) && "default" %in% names2(dots)) {
  warning("...")
  names(dots)[names(dots) == "default"] <- "defaults"
  return(do.call(modify_defaults, dots))
}
```

Reply:
In what way? Maybe we can also use `substitute(alist(...))` here to generate the correct call?
Possibly even modifying `match.call()` instead.

Reply:
OK, seems working now after pulling the naming logic into a helper.